kf_lib_data_ingest/etl/load/load_v2.py

```diff
@@ -0,0 +1,79 @@
+"""
+Version 2
+Module for loading the transform output into the dataservice. It converts the
+merged source data into complete message payloads according to a given API
+specification, and then sends those messages to the target server.
+"""
+from pprint import pformat
+
+from kf_lib_data_ingest.common import constants
+from kf_lib_data_ingest.etl.load.load_v1 import LoadStage as LoadV1
+
+
+class LoadStage(LoadV1):
+    def _get_target_id_from_record(self, entity_class, record):
+        """
+        Find the target service ID for the given record and entity class.
+
+        :param entity_class: one of the classes contained in the all_targets list
+        :type entity_class: class
+        :param record: a record of extracted data
+        :type record: dict
+        :return: the target service ID
+        :rtype: str
+        """
+        # check if target ID is given
+        tic = record.get(entity_class.target_id_concept)
+        if tic and (tic != constants.COMMON.NOT_REPORTED):
+            return tic
+
+        # check the cache
+        try:
+            key_components = self._do_target_get_key(entity_class, record)
+            tic = self._get_target_id_from_key(
+                entity_class.class_name, str(key_components)
+            )
```

Comment:
After this block, just

```python
return tic
```

Reply:
Then it would never check the server.

Reply:
Ah, the above logic is checking whether a key-components uid already exists.